doc-content/shared-kie-docs/src/main/asciidoc/KieServer/kie-server-rest-api-solvers-ref.adoc

```diff
@@ -0,0 +1,419 @@
+          }
+        },
+        {
+          "timeslot": { "-reference": "../../employeerostering.employeerostering.Shift/timeslot" },
+          "requiredSkill": {
+            "-reference": "../../../employeeList/employeerostering.employeerostering.Employee[2]/skills/employeerostering.employeerostering.Skill"
+          }
+        }
+      ]
+    },
+    "skillList": {
+      "employeerostering.employeerostering.Skill": [
+        { "-reference": "../../employeeList/employeerostering.employeerostering.Employee/skills/employeerostering.employeerostering.Skill" },
+        {
+          "-reference": "../../employeeList/employeerostering.employeerostering.Employee[3]/skills/employeerostering.employeerostering.Skill"
+        },
+        {
+          "-reference": "../../employeeList/employeerostering.employeerostering.Employee[2]/skills/employeerostering.employeerostering.Skill"
+        }
+      ]
+    },
+    "timeslotList": {
+      "employeerostering.employeerostering.Timeslot": { "-reference": "../../shiftList/employeerostering.employeerostering.Shift/timeslot" }
+    }
+  }
+}
+----
+
+.Example server response (JSON)
+[source,json]
+----
+
+----
+
+
+== [POST] /containers/{containerId}/solvers/{solverId}/state/terminating-early
+
+Stops the specified solver if it is running in the specified KIE container. This request does not delete the solver. The best solution can still be retrieved.
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`containerId`
+|ID of the KIE container where the solver resides
+|String
+|Required
+
+|`solverId`
+|ID of the solver
+|String
+|Required
+|===
+
+*<@Jiri: Example JSON response body here. Can't get a solver to start so can't stop one to see.>*
+
+.Example server response (JSON)
+[source,json]
+----
+
+----
+
+== [GET] /containers/{containerId}/solvers/{solverId}/bestsolution
+
+Returns the best solution from a specified solver within the specified KIE container at the time the request is made. If the solver is still running (with a current status of `SOLVING`), then this request returns the best solution found up to the point of the latest completion. Future requests after the solver completes might return a better solution.⁠
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`containerId`
+|ID of the KIE container where the solver resides
+|String
+|Required
+
+|`solverId`
+|ID of the solver
+|String
+|Required
+|===
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "container-id": "employee-rostering",
+  "solver-id": "solver1",
+  "solver-config-file": "employeerostering/employeerostering/EmployeeRosteringSolverConfig.solver.xml",
+  "status": "NOT_SOLVING",
+  "score": {
+    "value": null,
+    "scoreClass": null
+  },
+  "best-solution": "employeerostering.employeerostering.EmployeeRoster"
```

Comment:
I posted a problem in XStream and retrieved best solution as JSON. This is a sample respnose:
```json
{
  "container-id" : "employeerostering_1.0.0-SNAPSHOT",
  "solver-id" : "solver-json",
  "solver-config-file" : "employeerostering/employeerostering/EmployeeRosteringSolverConfig.solver.xml",
  "status" : "NOT_SOLVING",
  "score" : {
    "value" : "0hard/0soft",
    "scoreClass" : "org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore"
  },
  "best-solution" : {"employeerostering.employeerostering.EmployeeRoster":{
  "employeeList" : [ {
    "name" : "John",
    "skills" : [ {
      "name" : "reading"
    } ]
  }, {
    "name" : "Mary",
    "skills" : [ {
      "name" : "writing"
    } ]
  }, {
    "name" : "Petr",
    "skills" : [ {
      "name" : "speaking"
    } ]
  } ],
  "shiftList" : [ {
    "timeslot" : {
      "startTime" : {
        "year" : 2017,
        "month" : "JANUARY",
        "dayOfYear" : 1,
        "dayOfWeek" : "SUNDAY",
        "dayOfMonth" : 1,
        "monthValue" : 1,
        "hour" : 0,
        "minute" : 0,
        "second" : 0,
        "nano" : 0,
        "chronology" : {
          "calendarType" : "iso8601",
          "id" : "ISO"
        }
      },
      "endTime" : {
        "year" : 2017,
        "month" : "JANUARY",
        "dayOfYear" : 1,
        "dayOfWeek" : "SUNDAY",
        "dayOfMonth" : 1,
        "monthValue" : 1,
        "hour" : 1,
        "minute" : 0,
        "second" : 0,
        "nano" : 0,
        "chronology" : {
          "calendarType" : "iso8601",
          "id" : "ISO"
        }
      }
    },
    "requiredSkill" : {
      "name" : "reading"
    }
  }, {
    "timeslot" : {
      "startTime" : {
        "year" : 2017,
        "month" : "JANUARY",
        "dayOfYear" : 1,
        "dayOfWeek" : "SUNDAY",
        "dayOfMonth" : 1,
        "monthValue" : 1,
        "hour" : 0,
        "minute" : 0,
        "second" : 0,
        "nano" : 0,
        "chronology" : {
          "calendarType" : "iso8601",
          "id" : "ISO"
        }
      },
      "endTime" : {
        "year" : 2017,
        "month" : "JANUARY",
        "dayOfYear" : 1,
        "dayOfWeek" : "SUNDAY",
        "dayOfMonth" : 1,
        "monthValue" : 1,
        "hour" : 1,
        "minute" : 0,
        "second" : 0,
        "nano" : 0,
        "chronology" : {
          "calendarType" : "iso8601",
          "id" : "ISO"
        }
      }
    },
    "requiredSkill" : {
      "name" : "speaking"
    }
  }, {
    "timeslot" : {
      "startTime" : {
        "year" : 2017,
        "month" : "JANUARY",
        "dayOfYear" : 1,
        "dayOfWeek" : "SUNDAY",
        "dayOfMonth" : 1,
        "monthValue" : 1,
        "hour" : 0,
        "minute" : 0,
        "second" : 0,
        "nano" : 0,
        "chronology" : {
          "calendarType" : "iso8601",
          "id" : "ISO"
        }
      },
      "endTime" : {
        "year" : 2017,
        "month" : "JANUARY",
        "dayOfYear" : 1,
        "dayOfWeek" : "SUNDAY",
        "dayOfMonth" : 1,
        "monthValue" : 1,
        "hour" : 1,
        "minute" : 0,
        "second" : 0,
        "nano" : 0,
        "chronology" : {
          "calendarType" : "iso8601",
          "id" : "ISO"
        }
      }
    },
    "requiredSkill" : {
      "name" : "writing"
    }
  } ],
  "skillList" : [ {
    "name" : "reading"
  }, {
    "name" : "speaking"
  }, {
    "name" : "writing"
  } ],
  "timeslotList" : [ {
    "startTime" : {
      "year" : 2017,
      "month" : "JANUARY",
      "dayOfYear" : 1,
      "dayOfWeek" : "SUNDAY",
      "dayOfMonth" : 1,
      "monthValue" : 1,
      "hour" : 0,
      "minute" : 0,
      "second" : 0,
      "nano" : 0,
      "chronology" : {
        "calendarType" : "iso8601",
        "id" : "ISO"
      }
    },
    "endTime" : {
      "year" : 2017,
      "month" : "JANUARY",
      "dayOfYear" : 1,
      "dayOfWeek" : "SUNDAY",
      "dayOfMonth" : 1,
      "monthValue" : 1,
      "hour" : 1,
      "minute" : 0,
      "second" : 0,
      "nano" : 0,
      "chronology" : {
        "calendarType" : "iso8601",
        "id" : "ISO"
      }
    }
  } ],
  "dayOffRequestList" : [ ],
  "shiftAssignmentList" : [ ],
  "score" : "0hard/0soft"
}}
}
```

Reply:
I don't know how the timeslots are marshalled while the server seems to be unable to unmarshal them. The format of timeslot might change when we fix https://issues.jboss.org/browse/PLANNER-1356.

Reply:
Done. Implemented the example response.